### PR TITLE
Scrapper, Alisa Karpova - 22FPL2

### PR DESCRIPTION
**Pull request by AlisaKarpova**

### CODE REVIEW

File: lab_5_scrapper/scrapper.py [hunk 1]
```diff
@@ -176,6 +299,16 @@ def _fill_article_with_meta_information(self, article_soup: BeautifulSoup) -> No
         Args:
             article_soup (bs4.BeautifulSoup): BeautifulSoup instance
         """
+        self.article.title = article_soup.find('h1').text
+        date = article_soup.find(class_="blog-date").text
+        self.article.date = self.unify_date_format(date)
+        author = [''.join(article_soup.find(class_="article-author").text)]
+        if author:
+            self.article.author = author
```
Comment (dmitry-uraev) [inline lab_5_scrapper/scrapper.py:307]: should author be present as a list type? why not as a string? lint 305 - here you put author inside []

File: lab_5_scrapper/scrapper.py [hunk 2]
```diff
@@ -2,10 +2,46 @@
 Crawler implementation.
```
Comment (dmitry-uraev) [inline lab_5_scrapper/scrapper.py:2]: please, check https://github.com/fipl-hse/2023-2-level-ctlr/actions/runs/8720761911/job/23923160933?pr=25

you need to fix your import style

Comment (AlisaKarpova) [reply lab_5_scrapper/scrapper.py:2]: Fixed

**Comment by dmitry-uraev** [comment]
You need to parse author more carefully - see the error from tests attached
image

File: lab_5_scrapper/scrapper.py [hunk 3]
```diff
@@ -176,6 +314,15 @@ def _fill_article_with_meta_information(self, article_soup: BeautifulSoup) -> No
         Args:
             article_soup (bs4.BeautifulSoup): BeautifulSoup instance
         """
+        self.article.title = article_soup.find('h1').text
+        date = article_soup.find("time").get('datetime')
+        self.article.date = self.unify_date_format(str(date))
+        author = article_soup.find('p', class_="article-author").text
```
[the remark this answers is not in the document]
Comment (AlisaKarpova) [reply lab_5_scrapper/scrapper.py]: Fixed

File: lab_5_scrapper/scrapper.py [hunk 4]
```diff
@@ -127,11 +235,29 @@ def _extract_url(self, article_bs: BeautifulSoup) -> str:
         Returns:
             str: Url from HTML
         """
+        link = article_bs.find('a', class_="post-item-link")
+        if link:
+            return self.url_pattern + link.get('href')
+        return ''
```
Comment (AlisaKarpova) [inline lab_5_scrapper/scrapper.py:241]: I tried not to use a loop to extract the url but it extracts only one link and writes texts only from it. Where am I wrong, here or in find_articles?

Comment (dmitry-uraev) [reply lab_5_scrapper/scrapper.py:241]: you need to call _extract_url method several times in a loop from find_articles method 

each time you call _extract_url, you get one link and check it

Comment (AlisaKarpova) [reply lab_5_scrapper/scrapper.py]: Fixed

**Pull request closed without merging**